src/Compilers/CSharp/CscCore/csc.sh

```diff
@@ -0,0 +1,6 @@
+#!/bin/sh
+
+THISDIR=$(dirname $0)
+
+chmod +x $THISDIR/CoreRun 2>/dev/null
+$THISDIR/CoreRun $THISDIR/csc.exe $@
```

Comment:
In order to pass all command line arguments as they were passed in, you must use "$@" instead of $@.